jmix-flowui/flowui/src/main/java/io/jmix/flowui/facet/urlqueryparameters/GenericFilterUrlQueryParametersBinder.java

```diff
@@ -243,7 +247,7 @@ protected List<FilterComponent> deserializeConditions(List<String> conditionPara
     protected boolean isPermitted(DataLoader dataLoader, FilterComponent filterComponent) {
         if (filterComponent instanceof PropertyFilter<?> propertyFilter && propertyFilter.getProperty() != null) {
             MetaClass entityMetaClass = dataLoader.getContainer().getEntityMetaClass();
-            MetaPropertyPath propertyPath = entityMetaClass.getPropertyPath(propertyFilter.getProperty());
+            MetaPropertyPath propertyPath = metadataTools.resolveMetaPropertyPathOrNull(entityMetaClass, propertyFilter.getProperty());
```

Comment:
Remove explicit passing of `MetadataTools` in all constructors. It's a breaking change, because you change public API. You have `ApplicationContext` use it to obtain required bean. If a bean is intended to be used several times, make it lazy initialized, see `SingleFilterSupport` as an example.

Reply:
Fixed